example/components/ConnectWithSelect.tsx

```diff
@@ -52,9 +52,19 @@ export function ConnectWithSelect({
   error: Error | undefined
   setError: (error: Error | undefined) => void
 }) {
+  const chainIds = (() => {
+    switch (true) {
+      case connector instanceof Network:
+        return Object.keys(URLS)
+      case connector instanceof WalletConnect:
+        return Object.keys(MAINNET_CHAINS)
+      default:
+        return Object.keys(CHAINS)
+    }
+  })().map(Number)
```

Comment:
i don't love the inline anon function - you could make this a separate one that takes connector as an arg